src/sage/graphs/traversals.pyx

```diff
@@ -99,133 +300,15 @@ def _is_valid_lex_BFS_order(G, L):
-      resulting from the LexBFS traversal from vertex 0. This method assumes
-      that this array has already been allocated. However, there is no need to
-      initialize it.
-
-    - ``pred`` -- array of size ``n`` to store the predecessor of a vertex in
-      the LexBFS traversal from vertex 0. This method assumes that this array
-      has already been allocated and initialized (e.g., ``pred[i] = i``).
-
-    EXAMPLES:
-
-    Lex BFS ordering of the 3-sun graph::
-
-        sage: g = Graph([(1, 2), (1, 3), (2, 3), (2, 4), (2, 5), (3, 5), (3, 6), (4, 5), (5, 6)])
-        sage: g.lex_BFS(algorithm="fast")
-        [1, 2, 3, 5, 4, 6]
-    """
-    cdef uint32_t n = sd.n
-    cdef uint32_t n_slice = sd.m + 1
-    cdef MemoryAllocator mem = MemoryAllocator()
-    cdef uint32_t *sigma_inv = <uint32_t *>mem.allocarray(n, sizeof(uint32_t))
-    cdef uint32_t *slice_of = <uint32_t *>mem.allocarray(n, sizeof(uint32_t))
-    cdef uint32_t *slice_head = <uint32_t *>mem.allocarray(n_slice, sizeof(uint32_t))
-    cdef uint32_t *subslice = <uint32_t *>mem.allocarray(n_slice, sizeof(uint32_t))
-    cdef uint32_t i, j, k, l, a, old_k, v
-    cdef int wi
-
-    # Initialize slices (slice_of, slice_head, subslice) to 0
-    memset(slice_of, 0, n * sizeof(uint32_t))
-    slice_head[0] = 0
-    subslice[0] = 0
-
-    # Initialize the position of vertices in sigma
-    for i in range(n):
-        sigma[i] = i
-        sigma_inv[i] = i
-
-    k = 1
-    for i in range(n):
-        old_k = k
-        v = sigma[i]
-
-        # We update the labeling of all unordered neighbors of v
-        for wi in range(out_degree(sd, v)):
-            w = sd.neighbors[v][wi]
-            j = sigma_inv[w]
-            if j <= i:
-                # w has already been ordered
-                continue
-
-            # Get the name of the slice for position j
-            a = slice_of[j]
-            if slice_head[a] <= i:
-                # This slice cannot start at a position less than i
-                slice_head[a] = i + 1
-
-            # Get the position of the head of the slice
-            l = slice_head[a]
-            if l < n - 1 and slice_of[l + 1] == a:
-                if l != j:
-                    # Place w at the position of the head of the slice
-                    u = sigma[l]
-                    sigma_inv[u], sigma_inv[w] = j, l
-                    sigma[j], sigma[l] = u, w
-                    j = l
-                slice_head[a] += 1
-            if subslice[a] < old_k:
-                # Form a new slice
-                subslice[a] = k
-                slice_head[k] = j
-                subslice[k] = 0
-                k += 1
-
-            # Finally, we update the name of the slice for position j and set v
-            # as predecessor of w
-            slice_of[j] = subslice[a]
-            pred[w] = v
-
-
-def lex_BFS(G, reverse=False, tree=False, initial_vertex=None, algorithm="fast"):
+def lex_BFS(
```

Comment:
could be on a single line

Reply:
done in f7ad61fd650c56cb55570fd4fb64e59dfea10a3a